MAVProxy/mavproxy.py

```diff
@@ -936,8 +940,16 @@ def mkdir_p(dir):
 
 def log_writer():
     '''log writing thread'''
-    while not mpstate.status.exit:
-        mpstate.logfile_raw.write(bytearray(mpstate.logqueue_raw.get()))
+    print("Starting logs thread...")
+    while not mpstate.status.stop_event.is_set():
+        try:
+            bytes = mpstate.logqueue_raw.get(block=False)
+        except queue.Empty:
+            pass
+        else:
+            mpstate.logfile_raw.write(bytearray(bytes))
```

Comment:
Why make this change as part of this PR?

I'm not sure it's correct - `multiproc.Queue` doesn't necessarily throw `queue.Empty`.

Reply:
Easier approach - check `empty()` before access. Less pythonic, but more portable code.

Reply:
Actually: https://devguide.python.org/versions/
https://github.com/ArduPilot/MAVProxy/blob/e45946ae2b1a06e662f1cc5c176e2d07eab1e12d/MAVProxy/modules/lib/multiproc.py#L66

python3.7 is EOL, therefore the whole premise of PipeQueue is EOL based on the if statement.

Reply:
We don't kill user's setups just because the Python is not supported by the Python project.  Look at how long we supported Python2....